src/algebra/char_p/basic.lean

```diff
@@ -114,23 +115,36 @@ lemma eq_zero [char_zero R] : ring_char R = 0 := (eq R (char_p.of_char_zero R)).
 
 end ring_char
 
-theorem add_pow_char_of_commute [semiring R] {p : ℕ} [fact p.prime]
+variable {R}
+
+theorem add_pow_prime_eq_pow_add_pow_add_prime_mul_of_commute [semiring R] {p : ℕ}
```

Comment:
I pushed a golf.
Please move this to a more general file and put it in the `commute` namespace.
About the name, what do you think of `commute.exists_add_pow_prime_eq`?

Reply:
I tried moving it, but it needs both `data.nat.choose.dvd` and `data.nat.choose.sum`, neither of which import the other. What do you suggest?